File: frontend/lib/history.ts
/**
 * API client for negotiation history.
 *
 * Fetches the authenticated user's completed negotiation sessions
 * grouped by UTC day. Calls the Next.js catch-all proxy which
 * forwards to the backend GET /api/v1/negotiation/history endpoint.
 */

const API_BASE = "/api/v1";

// ---------------------------------------------------------------------------
// Interfaces — mirror backend Pydantic models exactly
// ---------------------------------------------------------------------------

export interface SessionHistoryItem {
  session_id: string;
  scenario_id: string;
  scenario_name: string;
  deal_status: string;
  total_tokens_used: number;
  token_cost: number;
  created_at: string;
  completed_at: string | null;
}

export interface DayGroup {
  date: string; // YYYY-MM-DD
  total_token_cost: number;
  sessions: SessionHistoryItem[];
}

export interface SessionHistoryResponse {
  days: DayGroup[];
  total_token_cost: number;
  period_days: number;
}

// ---------------------------------------------------------------------------
// API function
// ---------------------------------------------------------------------------

export async function fetchNegotiationHistory(
  email: string,
  days?: number,
): Promise<SessionHistoryResponse> {
  const params = new URLSearchParams({ email });
  if (days !== undefined) {
    params.set("days", String(days));
  }

  const res = await fetch(`${API_BASE}/negotiation/history?${params}`);

  if (!res.ok) {
    let detail: string;
    try {
      const body = await res.json();
      detail = body.detail ?? body.message ?? JSON.stringify(body);
    } catch {
      detail = res.statusText || `HTTP ${res.status}`;
    }
    throw new Error(detail);
  }

  return res.json();
}

File: backend/app/models/history.py
"""Pydantic V2 response models for the negotiation history endpoint.

Requirements: 2.1, 2.2, 2.3
"""

from pydantic import BaseModel, Field


class SessionHistoryItem(BaseModel):
    """A single completed negotiation session summary.

    Requirement 2.1: session_id, scenario_id, scenario_name, deal_status,
    total_tokens_used (ge=0), token_cost (ge=1), created_at, completed_at (optional).
    """

    session_id: str
    scenario_id: str
    scenario_name: str
    deal_status: str
    total_tokens_used: int = Field(ge=0)
    token_cost: int = Field(ge=1)
    created_at: str
    completed_at: str | None = None


class DayGroup(BaseModel):
    """Sessions grouped by UTC calendar day.

    Requirement 2.2: date (YYYY-MM-DD), total_token_cost (ge=0), sessions list.
    """

    date: str
    total_token_cost: int = Field(ge=0)
    sessions: list[SessionHistoryItem]


class SessionHistoryResponse(BaseModel):
    """Top-level history endpoint response.

    Requirement 2.3: days list, total_token_cost (ge=0), period_days (ge=1).
    """

    days: list[DayGroup]
    total_token_cost: int = Field(ge=0)
    period_days: int = Field(ge=1)
